From 72ba8eebe0f5ba9890055c7cead6b87cef21cbd3 Mon Sep 17 00:00:00 2001
From: Cole Robinson <crobinso@redhat.com>
Date: Mon, 8 Oct 2012 17:00:49 -0400
Subject: [PATCH] Add --cpuset=auto test confirming it actually works
Mail-Followup-To: rhvirt-patches@redhat.com

https://bugzilla.redhat.com/show_bug.cgi?id=834495

(cherry picked from commit e1e2b4ae44ceb9f9d0d397bdf78e7e7694a98ecd)
---
 tests/cli-test-xml/compare/cpuset-auto.xml | 52 ++++++++++++++++++++++++++++++
 tests/clitest.py                           |  5 +++
 2 files changed, 57 insertions(+)
 create mode 100644 tests/cli-test-xml/compare/cpuset-auto.xml

diff --git a/tests/cli-test-xml/compare/cpuset-auto.xml b/tests/cli-test-xml/compare/cpuset-auto.xml
new file mode 100644
index 0000000..fde679b
--- /dev/null
+++ b/tests/cli-test-xml/compare/cpuset-auto.xml
@@ -0,0 +1,52 @@
+<domain type='test'>
+  <name>foobar</name>
+  <uuid>00000000-1111-2222-3333-444444444444</uuid>
+  <memory>65536</memory>
+  <currentMemory>65536</currentMemory>
+  <vcpu cpuset='0,2,4,6,8,10,12,14'>2</vcpu>
+  <os>
+    <type arch='i686'>hvm</type>
+    <boot dev='network'/>
+  </os>
+  <features>
+    <acpi/><apic/><pae/>
+  </features>
+  <clock offset="utc"/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>destroy</on_reboot>
+  <on_crash>destroy</on_crash>
+  <devices>
+    <emulator>/usr/bin/test-hv</emulator>
+    <interface type='user'>
+      <mac address='00:11:22:33:44:55'/>
+    </interface>
+    <input type='mouse' bus='ps2'/>
+    <console type='pty'/>
+  </devices>
+</domain>
+<domain type='test'>
+  <name>foobar</name>
+  <uuid>00000000-1111-2222-3333-444444444444</uuid>
+  <memory>65536</memory>
+  <currentMemory>65536</currentMemory>
+  <vcpu cpuset='0,2,4,6,8,10,12,14'>2</vcpu>
+  <os>
+    <type arch='i686'>hvm</type>
+    <boot dev='network'/>
+  </os>
+  <features>
+    <acpi/><apic/><pae/>
+  </features>
+  <clock offset="utc"/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>restart</on_crash>
+  <devices>
+    <emulator>/usr/bin/test-hv</emulator>
+    <interface type='user'>
+      <mac address='00:11:22:33:44:55'/>
+    </interface>
+    <input type='mouse' bus='ps2'/>
+    <console type='pty'/>
+  </devices>
+</domain>
diff --git a/tests/clitest.py b/tests/clitest.py
index 7e9d400..0694159 100644
--- a/tests/clitest.py
+++ b/tests/clitest.py
@@ -78,6 +78,7 @@ clean_files = (new_images + exist_images +
 
 test_files = {
     'TESTURI'           : testuri,
+    'DEFAULTURI'        : "__virtinst_test__test:///default,predictable",
     'REMOTEURI'         : remoteuri,
     'KVMURI'            : kvmuri,
     'XENURI'            : xenuri,
@@ -564,6 +565,10 @@ args_dict = {
          "--security type=static,label='system_u:object_r:svirt_image_t:s0:c100,c200' "
          """ --numatune \\"1-3,5\\",mode=preferred """,
          "many-devices"),
+        # --cpuset=auto actually works
+        ("--connect %(DEFAULTURI)s --hvm --nodisks --pxe --cpuset auto "
+         "--vcpus 2",
+         "cpuset-auto"),
       ],
 
      }, # category "misc"
-- 
1.7.12.3

